gearcatalog/frontend/src/components/common/__tests__/Input.test.tsx

```diff
@@ -0,0 +1,60 @@
+import React from 'react';
+import { shallow } from 'enzyme';
+import Input from '../Input';
+
+beforeEach(() => {
+  jest.clearAllMocks();
+});
+
+describe('Input', () => {
+  test('Should render component and return div with text input', () => {
+    const props = {
+      name: 'foo',
+      val: 'bar',
+      type: 'text',
+      setVal: jest.fn(),
```

Comment:
For my own edification, why have you listed out the props in an object here instead of directly in the JSX?

Reply:
Typescript gets mad if you put test props directly into the component for whatever reason. So I put it into a object and spread it out into the component. If I don't do that my tests won't run.